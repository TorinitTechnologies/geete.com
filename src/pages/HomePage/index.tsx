import Button from '@atoms/Button';
import { makeUserLoggedOut } from '@utils/authUtils';

function HomePage() {
	const onLogout = () => {
		makeUserLoggedOut();
	};

	return (
		<div data-testid={'home_page'}>
			<Button
				id={'logout'}
				name={'logout'}
				dataTestId={'log_out_btn'}
				title={'Logout'}
				text={'Logout'}
				variant={'outlined'}
				onClick={onLogout}
			/>
		</div>
	);
}

export default HomePage;
