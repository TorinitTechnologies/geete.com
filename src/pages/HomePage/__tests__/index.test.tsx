import { fireEvent, render, screen } from '@testing-library/react';
import renderer from 'react-test-renderer';
import HomePage from '../index';

describe('HomePage', () => {
	it('renders correctly', () => {
		const tree = renderer.create(<HomePage />).toJSON();
		expect(tree).toMatchSnapshot();
	});

	it('HomePage should render', async () => {
		render(<HomePage />);
		const homePage = await screen.findByTestId('home_page');
		expect(homePage).toBeInTheDocument();
	});
	it('should have log out button', async () => {
		render(<HomePage />);
		const logOutButton = await screen.findByTestId('log_out_btn');
		expect(logOutButton).toBeInTheDocument();
	});
	it('should have log out button', async () => {
		render(<HomePage />);
		const logOutButton = await screen.findByTestId('log_out_btn');
		fireEvent.click(logOutButton);
	});
});
