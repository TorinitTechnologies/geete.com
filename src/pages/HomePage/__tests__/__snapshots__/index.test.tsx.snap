// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HomePage renders correctly 1`] = `
<div
  data-testid="home_page"
>
  <button
    className="MuiButtonBase-root MuiButton-root MuiButton-outlined MuiButton-outlinedPrimary MuiButton-sizeMedium MuiButton-outlinedSizeMedium MuiButton-root MuiButton-outlined MuiButton-outlinedPrimary MuiButton-sizeMedium MuiButton-outlinedSizeMedium sc-bdvumH css-1rwt2y5-MuiButtonBase-root-MuiButton-root"
    data-testid="log_out_btn"
    disabled={false}
    id="logout"
    name="logout"
    onBlur={[Function]}
    onClick={[Function]}
    onContextMenu={[Function]}
    onDragLeave={[Function]}
    onFocus={[Function]}
    onKeyDown={[Function]}
    onKeyUp={[Function]}
    onMouseDown={[Function]}
    onMouseLeave={[Function]}
    onMouseUp={[Function]}
    onTouchEnd={[Function]}
    onTouchMove={[Function]}
    onTouchStart={[Function]}
    tabIndex={0}
    title="Logout"
    type="button"
  >
    Logout
  </button>
</div>
`;
